examples/source/interactivity-dynamic-content/Dynamic_Menu_With_Sub_Menus.html

```diff
@@ -0,0 +1,167 @@
+  <!-- Import the `amp-mustache` component is required for managing our templates with `amp-list` -->
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+  <!-- Import the `amp-list` component to import the list of menu items -->
+  <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
+  <!-- Import the `amp-sidebar` component to manage the sidebar UI element -->
+  <script async custom-element="amp-sidebar" src="https://cdn.ampproject.org/v0/amp-sidebar-0.1.js" async=""></script>
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+
+  <style amp-custom>
+    #sideNav {
+      width: 100%;
+      height: 100%;
+      background-color: #fff;
+      max-width: 100vw;
+    }
+
+    .header-primary li a {
+      padding: 8px 8px 8px 32px;
+      text-decoration: none;
+      font-size: 16px;
+      color: #818181;
+      display: block;
+      transition: 0.3s;
+      border-bottom: 1px solid #e6e6e6;
+      font-family: "Open Sans", sans-serif;
+      font-size: 16px;
+      font-weight: 400;
+      line-height: 24px;
+      outline: 0;
+    }
+
+    .main-nav {
+      width: 100%;
+      height: 100%;
+      background-color: #fff;
+      max-width: 100vw;
+    }
+
+    .mobile-menu-header {
+      background-color: blue;
+      height: 45px;
+      text-align: center;
+      line-height: 45px;
+      color: white;
+    }
+
+    .closebtn {
+      top: 0px;
+      right: 15px;
+      outline: 0;
+      position: absolute;
+    }
+
+    .innermenulinks {
+      padding: 8px 8px 8px 32px;
+      text-decoration: none;
+      font-size: 16px;
+      color: #818181;
+      display: block;
+      transition: 0.3s;
+      border-bottom: 1px solid #e6e6e6;
+      font-family: "Open Sans", sans-serif;
+      line-height: 24px;
+      font-weight: 400;
+    }
+
+    .rightarrow:after {
+      content: ">";
+      float: right;
+      padding-right: 10px;
+      font-weight: bolder;
+      font-size: 25px;
+    }
+
+    .leftarrow:before {
+      content: "<";
+      float: left;
+      padding-left: 10px;
+      font-weight: bolder;
+      font-size: 25px;
+    }
+
+  </style>
+</head>
+<!-- ## Implementation -->
```

Comment:
You can directly annotate specific code blocks. In this case I would first annotate the button and explain that it will trigger the menu.